docs/reference/remap2/expressions/assignment.cue

```diff
@@ -0,0 +1,116 @@
+package metadata
+
+remap2: expressions: assignment: {
+	title: "Assignment"
+	description: """
+		An "assignment" expression assigns the result of the right-hand side expression to the left-hand side
+		target (path or variable).
+		"""
+	return: """
+		Returns the value of the right-hand side expression only if the expression succeeeds. If the expression errors,
+		then the error must be [handled](\(urls.vrl_errors_reference)) and null is returned.
+		"""
+
+	grammar: {
+		source: """
+			target ~ operator ~ expression
+			"""
+		definitions: {
+			target: {
+				description: """
+					The `target` must be a path,
+					with an optional second variable for error handling if the right-hand side is fallible.
+					"""
+			}
+			operator: {
+				description: """
+					The `operator` delimits the `target` and `expression` and defines assignment conditions.
+					"""
+				enum: {
+					"=": """
+						Simple assignment operator. Assigns the result from the left-hand side to the right-hand side:
+
+						```vrl
+						.field = "value"
+						```
+						"""
+					"??=": """
+						Assigns _only_ if the right hand side does not error. Useful when invoking fallible functions
+						on the right-hand side:
+
+						```vrl
+						.structured ??= parse_json(.message)
+						```
+						"""
+				}
+			}
+			expression: {
+				description: """
+					If the `target` is a variable, then the `expression` can be any	expression.
+
+					If the `target` is a path, then the `expression` can be any expression that returns a supported map
+					value type (ex: not a regular expression).
+					"""
+			}
+		}
+	}
+
+	examples: [
+		{
+			title: "Path assignment"
+			source: #"""
+				.message = "Hello, World!"
+				"""#
+			return: "Hello, World!"
+			output: log: message: "Hello, World!"
+		},
+		{
+			title: "Nested path assignment"
+			source: #"""
+				.parent.child = "Hello, World!"
+				"""#
+			return: "Hello, World!"
+			output: log: parent: child: "Hello, World!"
+		},
+		{
+			title: "Double assignment"
+			source: #"""
+				.first = .second = "Hello, World!"
+				"""#
+			return: "Hello, World!"
+			output: log: {
+				first:  "Hello, World!"
+				second: "Hello, World!"
+			}
+		},
+		{
+			title: "Array element assignment"
+			source: #"""
+				.array[1] = "Hello, World!"
+				"""#
+			return: "Hello, World!"
+			output: log: array: [null, "Hello, World!"]
+		},
+		{
+			title: "Variable assignment"
+			source: #"""
+				my_variable = "Hello, World!"
+				"""#
+			return: "Hello, World!"
+		},
+		{
+			title: "Fallible assignment (success)"
+			source: #"""
+				parsed, err = parse_json("{\"Hello\": \"World!\"}")
+				"""#
+			return: Hello: "World!"
+		},
+		{
+			title: "Fallible assignment (error)"
+			source: #"""
+				parsed, err = parse_json("malformed")
+				"""#
+			return: null
```

Comment:
The return value here isn't `null`, it's a string containing the error message.

Here's how it works:

- `parsed` is assigned `null` because the function failed
- `err` is assigned the error message
- the assignment expression returns the value assigned to `err` because the function failed, if it had succeeded, the expression would return the value assigned to `parsed`.

Reply:
Right, but the return value in this context represents the `parsed` variable, which will be null, right?